src/app/qgisapp.cpp

```diff
@@ -1690,6 +1699,9 @@ QgisApp::~QgisApp()
   // shouldn't be needed, but from this stage on, we don't want/need ANY map canvas refreshes to take place
   mFreezeCount = 1000000;
 
+  unregisterDevToolFactory( mNetworkLoggerWidgetFactory.get() );
```

Comment:
<hint you didn't ask for>
  Wouldn't that be better a custom deleter
</hint you didn't ask for>

Reply:
Possibly... but I want to control the order of destruction here... so, maybe?

Reply:
My first thought was that a std::unique_ptr is quite an advanced tool for the level of control you want to have here. The second one that if you use it, you should connect unregister to it's deleter. The third one that it's not really an issue ;)

Reply:
Fixed -- I couldn't get templatey goodness working direct with unique_ptr (can't find a clean way to specialize make_unique to apply custom logic in the constructor) so I've just gone with a custom RAII class instead.